docs/CMakeLists.txt

```diff
@@ -3,28 +3,26 @@ add_custom_target(docs)
 if(NOT Doxygen_FOUND)
   message(STATUS "Doxygen not found! building without doxygen docs.")
 else()
+  set(DOXYGEN_EXTRACT_PRIVATE YES)
 
+  set(DOXYGEN_EXTRACT_ALL YES)
+  set(DOXYGEN_BUILTIN_STL_SUPPORT YES)
+  set(DOXYGEN_RECURSIVE YES)
+  set(DOXYGEN_GENERATE_TREEVIEW YES)
 
-set(DOXYGEN_EXTRACT_PRIVATE YES)
-
-set(DOXYGEN_EXTRACT_ALL YES)
-set(DOXYGEN_BUILTIN_STL_SUPPORT YES)
-set(DOXYGEN_RECURSIVE YES)
-set(DOXYGEN_GENERATE_TREEVIEW YES)
-
-doxygen_add_docs(doxygen "${PROJECT_SOURCE_DIR}/src" "${PROJECT_SOURCE_DIR}/helper_functions" WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}" )
+  doxygen_add_docs(doxygen "${PROJECT_SOURCE_DIR}/src" "${PROJECT_SOURCE_DIR}/helper_functions" WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}" )
 
+  # only build latex if doxygen is installed otherwise add_dependencies(docs doxygen) will throw
```

Comment:
Well one build has both installed, one has only LaTeX installed. I guess we could add a "only doxygen" build too. But I kind of assumed that LaTeX would be used inside the documentation and thus be useless without doxygen. If that is not the case we should also remove this comment

Reply:
I see.
Up to now, Latex is used to build the userguide.pdf and nothing else. I removed the comment.